docs: adding docstrings for the SparseAEPipeline class

REETI-SARUP opened the pull request:
Fixes #59 

Signed-off-by: Reeti Sarup <reeti.sarup@gmail.com>

**Explain what this PR does:**
Adding docstrings for the `SparseAEPipeline` class & its methods in the `models/autoencoder/pipeline.py` file.

ab93 commented on numalogic/models/autoencoder/pipeline.py:
```diff
@@ -319,30 +319,127 @@ def with_model(
+    """
+
     def __init__(self, beta=1e-3, rho=0.05, method="kl_div", *args, **kwargs):
+        r"""
+        Function to initialize the data member values (including the inherited parent class members) when an object of this class is created. 
+
+        Args:
+            beta: regularization parameter (Defaults to 1e-3)
+            rho: sparsity parameter value (Defaults to 0.05)
+            method: regularization method (Defaults to "kl_div")
+            model: model instance
+            seq_len: sequence length
+            loss_fn: loss function used for training
+            optimizer: optimizer to be used for training
+            lr: learning rate
+            batch_size: batch size for training
+            num_epochs: number of epochs for training
+            std_tolerance: determines how many times the standard deviation to be used for threshold
+            reconerr_method: method used to calculate the distance between the original and the reconstucted data
+            threshold_min: the minimum threshold to use
+            resume_train: parameter to decide if resume training is needed
+        """
```
You can remove the docstring in __init__, since you already added that in the class docstring.

REETI-SARUP replied:
Sure, done.

ab93 reviewed (changes requested):
Thanks @REETI-SARUP. A minor comment. Also, can you run `make lint` locally, and push the changes? The linting check is failing.

REETI-SARUP commented:
> Thanks @REETI-SARUP. A minor comment. Also, can you run `make lint` locally, and push the changes? The linting check is failing.

Sure @ab93, thanks, I've addressed the review suggestion and pushed the linting fixes for `pipeline.py`. Could you please re-run the workflows / checks again on the PR to see if the linting check is passing now in the PR, it is currently showing that running the workflows / checks on the PR requires approval from maintainer of this repo.

ab93 reviewed (approved):
Looks great @REETI-SARUP ! Merging.
